src/hooks/useDataValidation.js

```diff
@@ -0,0 +1,44 @@
+import { useState, useEffect } from 'react';
+
+export const useDataValidation = (message, regex) => {
+	const [state, setState] = useState({
+		value: '',
+		error: false,
+		message: '',
+	});
+
+	const handleState = (event) => {
+		event.preventDefault();
+		const value = event.target.value;
+		const check = value.match(regex);
+		if (check) {
+			setState((prev) => ({
+				...prev,
+				value: value,
+				error: false,
+			}));
+			return true;
+		} else {
+			setState((prev) => ({
+				...prev,
+				message: message,
+				error: true,
+			}));
+			return false;
+		}
+	};
+
+	useEffect(() => {
+		if (state.error) {
+			setTimeout(() => {
+				setState((prev) => ({
+					...prev,
+					message: '',
+					error: false,
+				}));
+			}, 1000);
+		}
+	}, [state.error]);
+
+	return [handleState, state];
```

Comment:
so, basically, you have this hook as a cooler `useState`, so why don't you make the syntax similar? with the first param being value and the second being setter?

Reply:
In the useValidation custom hook, the syntax is changed to be similar to that of the ReactHooks. Obviously, it's better that way.